libafl/src/stages/gen.rs

```diff
@@ -0,0 +1,74 @@
+//! A [`Stage`] that generates a single input via a [`Generator`] and evaluates
+//! it using the fuzzer, possibly adding it to the corpus.
+
+use core::marker::PhantomData;
+
+use crate::{
+    generators::Generator,
+    inputs::UsesInput,
+    stages::Stage,
+    state::{HasCorpus, HasRand, UsesState},
+    Error, Evaluator,
+};
+
+/// A [`Stage`] that generates a single input via a [`Generator`] and evaluates
+/// it using the fuzzer, possibly adding it to the corpus.
+///
+/// This stage can be used to construct black-box (e.g., grammar-based) fuzzers.
+#[derive(Debug)]
+pub struct GenStage<Z, G>(G, PhantomData<Z>)
+where
+    Z: UsesState,
+    G: Generator<<<Z as UsesState>::State as UsesInput>::Input, Z::State>;
+
+impl<Z, G> GenStage<Z, G>
+where
+    Z: UsesState,
+    G: Generator<<<Z as UsesState>::State as UsesInput>::Input, Z::State>,
+{
+    /// Create a new [`GenStage`].
+    ///
+    /// The `Z` parameter is the fuzzer, and is not used: it only exists to
+    /// give type inference a helping hand.
+    pub fn new(g: G, _z: &Z) -> Self {
```

Comment:
you don't need _z here. I could compile it without this extra parameter. 
you'll pass this to fuzzer.fuzz_loop so it can tell the type right?

Reply:
Good point yeah, will remove